sql/sql_yacc.yy

```diff
@@ -450,7 +450,7 @@ void warn_about_deprecated_national(THD *thd)
   1. We do not accept any reduce/reduce conflicts
   2. We should not introduce new shift/reduce conflicts any more.
 */
-%expect 108
+%expect 109
```

Comment:
I guess here (and in the high priority DDL PR) no easy way to avoid this?

Reply:
We can fix this by reworking the grammar rules so that the thread id in option_type_with_thread_id is non-optional. It looks like we're confusing the `start_option_value_list_following_option_type` rule, because it doesn't know if it needs to reduce `option_type` to `option_type_with_thread_id`.

Reply:
@inikep, as per @lth's suggestion, can something like the following grammar rules be used instead and can the parse tree be able to support it easily? It basically removes option_type from option_type_with_thread_id and explicitly adds productions for option_type_with_thread_id to the appropriate locations that have productions for option_type. Changes to your existing rules would be:

```
option_type_with_thread_id:
          LOCAL_SYM opt_var_tid 
        | SESSION_SYM opt_var_tid
        ;

start_option_value_list_following_option_type:
          option_type option_value_following_option_type option_value_list_continued
        | option_type_with_thread_id option_value_following_option_type option_value_list_continued
        | option_type TRANSACTION_SYM transaction_characteristics
        ;

option_value:
          option_type option_value_following_option_type
        | option_type_with_thread_id option_value_following_option_type
        | option_value_no_option_type
        ;
```

Note I'm only including the grammar rules here without any code associated on those rules. The above change, when compiled with bison, does not have any additional shift/reduce conflicts.

Reply:
Done